cardano-api/test/Test/Cardano/Api/Typed/Gen.hs

```diff
@@ -105,28 +118,110 @@ genStakeCredential = do
   vKey <- genVerificationKey AsStakeKey
   return . StakeCredentialByKey $ verificationKeyHash vKey
 
-{-
---TODO: Currently undefined
-genTxBodyByron :: TxBody Byron
-genTxBodyByron = getTxBody <$> genTxByron
-
-genTxBodyShelley :: TxBody Shelley
+genTxBodyShelley :: Gen (TxBody Shelley)
 genTxBodyShelley =
-  Gen.choice
-    [ genTxBodyByron
-    , getTxBody <$> genTxShelley
-    ]
+   makeShelleyTransaction
+     <$> genTxExtraContent
+     <*> genTTL
+     <*> genTxFee
+     <*> Gen.list (Range.constant 1 10) genTxIn
+     <*> Gen.list (Range.constant 1 10) genShelleyTxOut
+
+genByronTxOut :: Gen (TxOut Byron)
+genByronTxOut =
+  TxOut <$> genAddressByron <*> genLovelace
+
+genShelleyTxOut :: Gen (TxOut Shelley)
+genShelleyTxOut =
+  TxOut <$> genAddressShelley <*> genLovelace
+
+genShelleyHash :: Gen (Crypto.Hash Crypto.Blake2b_256 ())
+genShelleyHash = return $ Crypto.hash ()
+
+genSlotNo :: Gen SlotNo
+genSlotNo = SlotNo <$> Gen.word64 Range.constantBounded
+
+-- TODO: Should probably have a naive generator that generates no inputs, no outputs etc
```

Comment:
Is this TODO saying we should have an additional generator?

Reply:
Yep, so we may want to have a naive generator that generates no transactions inputs (which would be an invalid transaction) for other testing purposes. It's okay to not have right now.